### PR TITLE
Sell crypto option

### PR DESCRIPTION
**Pull request by OK1ez**
-Adds config option for selling crypto

https://streamable.com/jkgitx
https://imgur.com/a/yGKzmBj

### CODE REVIEW

**Comment by Smokiiee** [comment]
did you miss sell crypto in crypto.js ? html is sending to crypto-send-sell but there is nothing like that in crypto.js

**Comment by OK1ez** [comment]
> did you miss sell crypto in crypto.js ? html is sending to crypto-send-sell but there is nothing like that in crypto.js

Should be fixed now, went a little bit fast when i was transferring the files from my edited version to a fresh install

**Comment by Smokiiee** [comment]
> > did you miss sell crypto in crypto.js ? html is sending to crypto-send-sell but there is nothing like that in crypto.js
> 
> Should be fixed now, went a little bit fast when i was transferring the files from my edited version to a fresh install

i understand :) Testing now :D

**Comment by Smokiiee** [comment]
> > > did you miss sell crypto in crypto.js ? html is sending to crypto-send-sell but there is nothing like that in crypto.js
> > 
> > 
> > Should be fixed now, went a little bit fast when i was transferring the files from my edited version to a fresh install
> 
> i understand :) Testing now :D

works perfect :)

File: server/crypto.lua [hunk 1]
```diff
@@ -104,6 +104,35 @@ RegisterNetEvent('qb-phone:server:PurchaseCrypto', function(type, amount)
...
+
+        if Config.RenewedBanking then
+            local cid = Player.PlayerData.citizenid
+            local name = ("%s %s"):format(Player.PlayerData.charinfo.firstname, Player.PlayerData.charinfo.lastname)
+            exports['Renewed-Banking']:handleTransaction(cid, "Sold Crypto", cryptoAmount, txt, "Los Santos Crypto", name, "deposit")
+        end
+
+        RemoveCrypto(src, type, amount)
+end)
+
```
Comment (FjamZoo) [inline server/crypto.lua:135]: Please change the following

```suggestion
RegisterNetEvent('qb-phone:server:SellCrypto', function(type, amount)
    local src = source
    local Player = QBCore.Functions.GetPlayer(src)
    if not Player or not Player.PlayerData.metadata.crypto[type] then return end -- if the crypto dosnt exist
    local v = Config.CryptoCoins[GetConfig(type)]
    local cryptoAmount = tonumber(amount) * v.value

    if not v.sell then return end -- Only modders should be only to do this so no need to send a message to client
    
    local txt = "Sold " .. amount .. "x " .. v.abbrev
    
    if not RemoveCrypto(src, type, amount) then return end

        Player.Functions.AddMoney('bank', cryptoAmount, txt)
        TriggerClientEvent('qb-phone:client:CustomNotification', src,
            "WALLET",
            "You Sold "..amount.." "..type.."!",
            "fas fa-chart-line",
            "#D3B300",
            7500
        )

        if Config.RenewedBanking then
            local cid = Player.PlayerData.citizenid
            local name = ("%s %s"):format(Player.PlayerData.charinfo.firstname, Player.PlayerData.charinfo.lastname)
            exports['Renewed-Banking']:handleTransaction(cid, "Sold Crypto", cryptoAmount, txt, "Los Santos Crypto", name, "deposit")
        end
end)

```

Comment (FjamZoo) [reply server/crypto.lua:135]: Ok i do not know why tf it looks like that but pretty much just add 

`if not RemoveCrypto(src, type, amount) then return end`

Before u add the money this way we make sure that the client do not get money before any crypto is removed (security)

**Pull request merged**